diff --git a/qtbase/src/plugins/platforms/windows/qwindowsmousehandler.cpp b/qtbase/src/plugins/platforms/windows/qwindowsmousehandler.cpp
index acb6925..36905e1 100644
--- a/qtbase/src/plugins/platforms/windows/qwindowsmousehandler.cpp
+++ b/qtbase/src/plugins/platforms/windows/qwindowsmousehandler.cpp
@@ -271,6 +271,12 @@ bool QWindowsMouseHandler::translateMouseEvent(QWindow *window, HWND hwnd,
     QWindow *currentWindowUnderMouse = platformWindow->hasMouseCapture() ?
         QWindowsScreen::windowAt(globalPosition, CWP_SKIPINVISIBLE | CWP_SKIPTRANSPARENT) : window;
 
+    if (!currentWindowUnderMouse) {
+        QRect clientRect(QPoint(0, 0), window->size());
+        if (clientRect.contains(winEventPosition))
+            currentWindowUnderMouse = window;
+    }
+
     compressMouseMove(&msg);
     // Qt expects the platform plugin to capture the mouse on
     // any button press until release.
